File: Frontend/src/app/Classes/SocialPage.ts
import { SocialMediaType } from "./SocialMediaType";
import { SocialOperation } from "./SocialOperation";
import { User } from "./User";
export class SocialPage {
  id?: number; // Mark the 'id' property as optional with the '?'
  UserId!: any;
  pageLink!: string;
  socialMediaType!: SocialMediaType;
  socialOperation!: SocialOperation;
  pointsToSpend!: number;
}

File: Backend/src/main/java/com/example/stagefinal/entities/SocialPage.java
package com.example.stagefinal.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Setter
@Getter
@Entity
public class SocialPage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;


    @ManyToOne
    @JoinColumn(name = "user_id") // Specify the foreign key column name
    @JsonBackReference
    private User user;

    private String pageLink;


    @Enumerated(EnumType.STRING)
    private SocialMediaType socialMediaType;

    @Enumerated(EnumType.STRING)
    private SocialOperation socialOperation;

    private int pointsToSpend;

}
